sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ReadFromKafkaDoFn.java

```diff
@@ -407,4 +408,13 @@ public double getTotalSize(double numRecords) {
       return avgRecordSize.get() * numRecords / (1 + avgRecordGap.get());
     }
   }
+
+  private static Instant ensureTimestampWithinBounds(Instant timestamp) {
```

Comment:
Why do we need this function? Is it possible to have a timestamp out of window bounds?

Reply:
The MIN_TIMESTAMP produced in Dataflow is smaller than `BoundedWindow.MIN_TIMESTAMP`.